dandiapi/api/views/asset.py

```diff
@@ -85,7 +86,15 @@ def create(self, request, version__dandiset__pk, version__version):
             metadata=asset_metadata,
             version=version,
         )
-        asset.save()
+        try:
+            asset.save()
+        except IntegrityError as e:
+            # https://stackoverflow.com/questions/25368020/django-deduce-duplicate-key-exception-from-integrityerror
+            # https://www.postgresql.org/docs/13/errcodes-appendix.html
+            # Postgres error code 23505 == unique_violation
+            if e.__cause__.pgcode == '23505':
+                return Response('asset already exists', status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Minor but since it is an API response: 
```suggestion
                return Response('Asset Already Exists', status=status.HTTP_400_BAD_REQUEST)
```